### ISSUE TITLE
Since v0.8.2 `@dr.pogodin/react-native-static-server` requires `@dr.pogodin/react-native-fs` as a peer dependency, and it is incompatible with the original `react-native-fs` library in the host project

### ISSUE THREAD

**Issue by qwertynik** [birdofpreyru/react-native-static-server#65]
On using the static server, the following error comes up

<img width="914" alt="image" src="https://github.com/birdofpreyru/react-native-static-server/assets/6836536/572946d8-7ba9-42de-bbf9-0cd69375e04f">

Any ideas why this error is coming up? The recommended steps to resolve the error are followed, but still the error persists. 

On further searching, come across this issue on [react-native-fs](https://github.com/itinance/react-native-fs/issues/1197) where you are shared your fork. Is there anything that needs to be done to get the library working?

**Comment by qwertynik** [comment]
Read through the Releases page and found that 0.8.2 started using @dr.pogodin/react-native-fs. 
Installed 0.8.1 and this issue seems to have gone.

**Comment by exotexot** [comment]
Same.

**Comment by birdofpreyru** [comment]
Yes guys, it turns out React Native just does not link indirect native dependencies coming from RN libraries (https://github.com/react-native-community/cli/issues/870), and although there is some work in that direction, for now such dependencies should be explcitly installed into the host codebase (https://github.com/callstack/react-native-builder-bob#how-do-i-add-a-react-native-library-containing-native-code-as-a-dependency-in-my-library).

Thus, as a part of installing the latest RN Static Server you should also `npm install --save @dr.pogodin/react-native-fs`. I believe, it should not conflict with the old RNFS version, if you need both installed currently (because my fork still does not implement all functionality), as technically my fork and old lib register themselves as differently named native modules.

It is on me to update README, and `package.json` of RN Static Server to prevent users bumping into this issue.

**Comment by exotexot** [comment]
@birdofpreyru After installing 0.8.3 using the recommended "npx install-peerdeps @dr.pogodin/react-native-static-server", then running pod install, makes it on iOS and Catalyst unable to build. Due to:

"45 duplicate symbols for architecture x86_64" ... I guess running the "Original" RNFS and your fork of it in parallel is problematic.


```
Ld /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/cat724.app/Contents/MacOS/cat724 normal (in target 'cat724' from project 'cat724')
    cd /Users/vinzenz/Repositories/cat724/ios
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -Xlinker -reproducible -target x86_64-apple-ios13.1-macabi -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-maccatalyst -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/lib/swift -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/usr/lib/swift -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/lib/swift -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L\ /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/usr/lib\ /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/maccatalyst -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/BEMCheckBox -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/CocoaAsyncSocket -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/DoubleConversion -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Flipper -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Flipper-Boost-iOSX -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Flipper-Fmt -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Flipper-Folly -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Flipper-PeerTalk -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/FlipperKit -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RCT-Folly -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RCTTypeSafety -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNCAsyncStorage -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNCCheckbox -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNCMaskedView -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNDeviceInfo -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFastImage -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNGestureHandler -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNPdftron -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNReanimated -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNSVG -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNScreens -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNVectorIcons -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-Codegen -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-Core -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-CoreModules -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-NativeModulesApple -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTAnimation -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTAppDelegate -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTBlob -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTImage -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTLinking -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTNetwork -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTSettings -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTText -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-RCTVibration -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-cxxreact -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-debug -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-hermes -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-jsi -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-jsiexecutor -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-jsinspector -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-logger -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-perflogger -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-runtimescheduler -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/React-utils -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/ReactCommon -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/SDWebImage -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/SDWebImageWebPCoder -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/SocketRocket -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/Yoga -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/YogaKit -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-static-server -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/fmt -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/glog -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/libevent -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/libwebp -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-blur -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-netinfo -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-safe-area-context -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-segmented-control -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-sensitive-info -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/react-native-webview -L/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/toolbar-android -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/lib/swift -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/usr/lib -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/maccatalyst -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/EagerLinkingTBDs/Debug-maccatalyst -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst -F/Users/vinzenz/Repositories/cat724/ios/Pods/Flipper-DoubleConversion/Frameworks -F/Users/vinzenz/Repositories/cat724/ios/Pods/Flipper-Glog/Frameworks -F/Users/vinzenz/Repositories/cat724/ios/Pods/OpenSSL-Universal/Frameworks -F/Users/vinzenz/Repositories/cat724/ios/Pods/PDFNet -F/Users/vinzenz/Repositories/cat724/ios/Pods/hermes-engine/destroot/Library/Frameworks/universal -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/Flipper-DoubleConversion -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/Flipper-Glog -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/OpenSSL-Universal -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/PDFNet -F/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/hermes-engine/Pre-built -FPods/PDFNet -FPods/PDFNet/PDFNet.xcframework -FPods/PDFNet/Tools-Localization -FPods/PDFNet/Tools.xcframework -FPods/PDFNet/PDFNet.xcframework/ios-arm64 -FPods/PDFNet/PDFNet.xcframework/ios-arm64_x86_64-maccatalyst -FPods/PDFNet/PDFNet.xcframework/ios-arm64_x86_64-simulator -FPods/PDFNet/Tools.xcframework/ios-arm64 -FPods/PDFNet/Tools.xcframework/ios-arm64_x86_64-maccatalyst -FPods/PDFNet/Tools.xcframework/ios-arm64_x86_64-simulator -iframework /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/System/Library/Frameworks -iframework /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/System/Library/Frameworks -filelist /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/cat724.build/Debug-maccatalyst/cat724.build/Objects-normal/x86_64/cat724.LinkFileList -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @loader_path/../Frameworks -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -dead_strip -no_dead_strip_inits_and_terms -Xlinker -object_path_lto -Xlinker /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/cat724.build/Debug-maccatalyst/cat724.build/Objects-normal/x86_64/cat724_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -stdlib\=libc++ -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L/System/iOSSupport/usr/lib/swift -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/cat724.build/Debug-maccatalyst/cat724.build/Objects-normal/x86_64/cat724.swiftmodule -ObjC -lBEMCheckBox -lCocoaAsyncSocket -lDoubleConversion -lFlipper -lFlipper-Boost-iOSX -lFlipper-Fmt -lFlipper-Folly -lFlipper-PeerTalk -lFlipperKit -lRCT-Folly -lRCTTypeSafety -lRNCAsyncStorage -lRNCCheckbox -lRNCMaskedView -lRNDeviceInfo -lRNFS -lRNFastImage -lRNGestureHandler -lRNPdftron -lRNReanimated -lRNSVG -lRNScreens -lRNVectorIcons -lReact-Codegen -lReact-Core -lReact-CoreModules -lReact-NativeModulesApple -lReact-RCTAnimation -lReact-RCTAppDelegate -lReact-RCTBlob -lReact-RCTImage -lReact-RCTLinking -lReact-RCTNetwork -lReact-RCTSettings -lReact-RCTText -lReact-RCTVibration -lReact-cxxreact -lReact-debug -lReact-hermes -lReact-jsi -lReact-jsiexecutor -lReact-jsinspector -lReact-logger -lReact-perflogger -lReact-runtimescheduler -lReact-utils -lReactCommon -lSDWebImage -lSDWebImageWebPCoder -lSocketRocket -lYoga -lYogaKit -lc++ -lc++abi -ldr-pogodin-react-native-fs -ldr-pogodin-react-native-static-server -lfmt -lglog -licucore -llibevent -llibwebp -lreact-native-blur -lreact-native-netinfo -lreact-native-safe-area-context -lreact-native-segmented-control -lreact-native-sensitive-info -lreact-native-webview -lstdc++ -ltoolbar-android -lz -framework AVFoundation -framework AudioToolbox -framework CFNetwork -framework CoreGraphics -framework CoreImage -framework CoreMedia -framework CoreText -framework Foundation -framework ImageIO -framework JavaScriptCore -framework MediaPlayer -framework MobileCoreServices -framework OpenSSL -framework PDFNet -framework Photos -framework QuartzCore -framework Security -framework SystemConfiguration -framework Tools -framework UIKit -framework double-conversion -framework glog -framework hermes -ObjC -lc++ -lPods-cat724 -Xlinker -no_adhoc_codesign -Xlinker -dependency_info -Xlinker /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Intermediates.noindex/cat724.build/Debug-maccatalyst/cat724.build/Objects-normal/x86_64/cat724_dependency_info.dat -o /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/cat724.app/Contents/MacOS/cat724

ld: warning: directory not found for option '-L /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/System/iOSSupport/usr/lib /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/maccatalyst'
ld: warning: option -no_dead_strip_inits_and_terms is obsolete and being ignored
ld: warning: dylib (/Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/XCFrameworkIntermediates/hermes-engine/Pre-built/hermes.framework/hermes) was built for newer Mac Catalyst version (14.0) than being linked (13.1)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._discretionary' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._backgroundTimeout' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._readTimeout' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._headers' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._params' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_CLASS_$_RNFSDownloadParams' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_METACLASS_$_RNFSDownloadParams' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._progressDivider' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_CLASS_$_RNFSDownloader' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_METACLASS_$_RNFSDownloader' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._lastProgressEmitTimestamp' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._session' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._bytesWritten' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._fromUrl' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._progressInterval' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._task' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._progressCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._errorCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._beginCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._completeCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._resumableCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._contentLength' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._lastProgressValue' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._toFile' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._fileHandle' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._cacheable' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._statusCode' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloadParams._background' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSDownloader._resumeData' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Downloader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Downloader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._binaryStreamOnly' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._headers' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploader._params' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_CLASS_$_RNFSUploadParams' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_METACLASS_$_RNFSUploadParams' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._files' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._fields' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_CLASS_$_RNFSUploader' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_METACLASS_$_RNFSUploader' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._toUrl' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploader._task' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._progressCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._errorCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._beginCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._completeCallback' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
duplicate symbol '_OBJC_IVAR_$_RNFSUploadParams._method' in:
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/RNFS/libRNFS.a(Uploader.o)
    /Users/vinzenz/Library/Developer/Xcode/DerivedData/cat724-dfptzowhzhtqnectpdqsttwrxxxx/Build/Products/Debug-maccatalyst/dr-pogodin-react-native-fs/libdr-pogodin-react-native-fs.a(Uploader.o)
ld: 45 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```


![Screenshot 2023-09-03 at 11 51 57](https://github.com/birdofpreyru/react-native-static-server/assets/10153413/52a67c7e-9e66-4b4c-a717-a5e49ba2b0d8)

**Comment by birdofpreyru** [comment]
Ogh… I see, to make it work for iOS / Mac I’d need to rename all top-level symbols in Objective-C code, or to wrap them in some namespace, to ensure no name clashes. Do you really need both new and old fs libs? Are you using those download & upload functionality in your app?

**Comment by exotexot** [comment]
I tried completely migrating to your RNFS fork and then realised that 

> Are you using those download & upload functionality in your app?

Upload + Download aren't working yet ^^.

I would also need the copyFolder function, which avmoroz specifically integrated upon my request as recursive copying of an entire folder on Windows took forever back then. 

Idk the state of your branch for this function, but I guess having both libraries in parallel would be the best for me for now. In my
App I was using an own fork of RNFS which integrated 3-4 other PRs from user which never were merged

**Comment by exotexot** [comment]
@birdofpreyru For now I can also just stay on 0.8.1 on macOS.

**Comment by birdofpreyru** [comment]
> _Idk the state of your branch for this function_

Verifying `copyFolder()` works isn't difficult; download & upload functionality is tricky &mdash; each includes a bunch of methods to verify, and testing them will be more cumbersome than other functions that are dedicated to a single operation and not having to interplay with each other.

> _For now I can also just stay on 0.8.1 on macOS._

Yeah... I guess, you should be fine with v0.8.1 for time being &mdash; the primary difference of later versions is that they work on Windows with the latest RN versions, thanks to the migration to my RNFS fork, and attempting to use the original RNFS with the latest RN should crash on Windows anyway &mdash; thus, no real benefits of running my and the original RNFS side-by-side and upgrading to the latest static-server version.

**Comment by birdofpreyru** [comment]
So, I guess, it is blocked until I have my RNFS version fully compatible with the original one, functionality-wise.

**Comment by exotexot** [comment]
Just out of curiosity: which part of the original RNFS wasn't working for you?

Because I can build my fork of RNFS just fine on windows. 

I'm using downlaodFile, uploadFile, readFile, writeFile, stat, copyFile, copyFolder

**Comment by birdofpreyru** [comment]
For me the original RNFS on Windows, with the latest RN, was just crashing the production build of Static Server's Example app on start-up. Do you mean it is not a problem with your fork in your app?

Anyway, when upgrading my fork of RNFS to the new RN architecture, there is a need to change lots of function signatures to ensure they correctly linked with JS layer, and some argument types have to change, _etc._ So, it needs time, even if I am not touching internal implementation of the stuff at the moment. And also while on this I am writing tests and revising documentation for each function &mdash; there is a bunch of undocumented behavior nuances, inconsistent between platforms, which wanna note, and potentially fix in the future.

From your list I have not yet checked `downloadFile()`, `uploadFile()`, and `copyFolder()`.

**Comment by konami93** [comment]
Hi @birdofpreyru ,Im having this error running on windows
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':dr.pogodin_react-native-fs'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':dr.pogodin_react-native-fs:classpath'.
      > Could not find com.android.tools.build:gradle:7.2.1.
        Searched in the following locations:
          - https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/7.2.1/gradle-7.2.1.pom
          - https://repo.maven.apache.org/maven2/com/android/tools/build/gradle/7.2.1/gradle-7.2.1.pom
        Required by:
            project :dr.pogodin_react-native-fs

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 8s

**Comment by birdofpreyru** [comment]
@konami93 this probably should be a separate issue, and probably in the https://github.com/birdofpreyru/react-native-fs repo. I personally use Ubuntu for RN builds targeting Android; and the last time I tried to test Android builds on Windows, that did not work, at least for the new RN architecture, due to bugs in RN itself [#20]. I saw RN team is doing some stuff to fix builds on Windows, so probably once they claim it is fixed, I'll have a look whether it works for this library.

**Comment by konami93** [comment]
I think I just managed it in windows, but I have a question @birdofpreyru. I can use @dr.pogodin/react-native-static-server to run a nodejs server from a script.js

**Comment by birdofpreyru** [comment]
Hey @konami93 , I am not sure I understood this

> _I can use @dr.pogodin/react-native-static-server to run a nodejs server from script.js_

But probably this is not what this library does, this library runs an HTTP server, specifically [Lighttpd](https://www.lighttpd.net), to serve assets to, say, an internal webview in the host app, thus allowing, for example, to locally run an web app inside an RN app. If this is what you need &mdash; you are in the right place; if otherwise you are looking to emulate Node environment inside an RN app &mdash; this is not what this project is about.

**Comment by birdofpreyru** [comment]
FYI @exotexot , I just released [@dr.pogodin/react-native-fs](https://www.npmjs.com/package/@dr.pogodin/react-native-fs) **v2.21.0-alpha.4**, in which `copyFolder()`, `downloadFile()`, and `uploadFiles()` functions work (presumably). I only have tested their base functionality, thus no guarantees that all options for download / upload, and other related functions, work (and work on all platforms). I guess, it now covers all functions you use, so maybe test it sometime, and let me know, if something is off.

**Comment by birdofpreyru** [comment]
Will be closed after upgrade to [@dr.pogodin/react-native-fs@2.22.0](https://github.com/birdofpreyru/react-native-fs/releases/tag/v2.22.0).
